src/main/java/uk/gov/ch/developer/docs/controller/developer/SignInController.java

```diff
@@ -9,107 +9,90 @@
-        this.redirectUri = reader.getMandatoryString("OAUTH2_REDIRECT_URI");
+    private SessionFactory sessionFactory;
+
+    public SignInController() {
+    }
+
+    private static String getRequestURL(final HttpServletRequest request) {
+        // Find the original requested url
+        final StringBuilder originalRequestUrl = new StringBuilder(
+                request.getRequestURL());
+        final String queryString = request.getQueryString();
+        if (queryString != null) {
+            originalRequestUrl.append("?").append(queryString);
+        }
+        return originalRequestUrl.toString();
     }
 
     @GetMapping
-    public void getSignIn(HttpServletRequest httpServletRequest,
-            HttpServletResponse httpServletResponse) throws IOException {
+    public void getSignIn(final HttpServletRequest httpServletRequest,
+            final HttpServletResponse httpServletResponse) throws IOException {
 
-        Session chSession = (Session) httpServletRequest
+        final Session chSession = (Session) httpServletRequest
                 .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
-
         // Redirect for user authentication (no scope specified)
-        redirectForAuth(chSession, httpServletRequest, httpServletResponse, null, false);
+        redirectForAuth(chSession, httpServletRequest, httpServletResponse);
     }
 
     /**
      * Redirects to a URI for the user to authenticate themselves
-     * 
+     *
      * @param session The user's session, retrieved from context
-     * @param request
-     * @param response
-     * @param companyNumber The company number a user is viewing
-     * @param force Boolean determining whether to force re-authentication
-     * @throws IOException
      */
-    protected void redirectForAuth(Session session, HttpServletRequest request,
-            HttpServletResponse response, String companyNumber, boolean force) throws IOException {
+    protected void redirectForAuth(final Session session, final HttpServletRequest request,
+            final HttpServletResponse response)
+            throws IOException {
 
-        // Find the original requested url
-        StringBuilder originalRequestUrl = new StringBuilder(request.getRequestURL().toString());
-        String queryString = request.getQueryString();
-        if (queryString != null) {
-            originalRequestUrl.append("?").append(queryString);
-        }
+        final String originalRequestUrl = getRequestURL(request);
+        final String nonce = generateSessionNonce(session);
 
-        // Set the scope
-        String scope = null;
-        if (companyNumber != null) {
-            scope = OAUTH_COMPANY_SCOPE_PREFIX + companyNumber;
-        }
+        // Build oauth uri and redirect
+        final String state = oauth.oauth2EncodeState(originalRequestUrl, nonce, "content");
+        final String authoriseUri = identityProvider.getAuthorisationUrl(state);
```

Comment:
I've noticed the `createAuthoriseURIWithForceAndHint` method has been removed. From my last conversations with @macreman this was decided to be left in for future-proofing as this was related to protecting the session from being hijacked etc.

Reply:
Ah, I was unaware of that conversation. I'll see about resurrecting it.

Reply:
Come to think about it, we could potentially use the Hijack filter which would implement this anyway, so we may not need it after all. I will need to check with @macreman on this.

Reply:
If it's not needed, then it should be removed.

Reply:
I have spoken to Billy and we may need to keep this in after all, I will go over it in detail with him tomorrow.